.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
```

Comment:
This repo provides lots of hooks.
I think we should adopt the hooks like `requirements-txt-fixer`, `file-contents-sorter`, `double-quote-string-fixer`. `check-yaml`, `pretty-format-json`, and `check-json`.
There are more of them but I think these were relatively more relevant.

Reply:
I like your list and the hooks you've already added. I think a couple more hooks of potential interest are `name-tests-test` and `debug-statements`

Reply:
@tommy-waltmann Thanks for the review.
I'll try to add them locally and check whether it catches any error or not.

Reply:
@tommy-waltmann, the `name-tests-test` hook checks whether the name of all the files in the folder `tests/` starts with `test_` or not.
Given this I think it may restrict the development as in signac-flow, we have python modules named `define_test_project.py`, `extract_templates.py` in the `tests/` folder.

Reply:
@tommy-waltmann, the `name-tests-test` hook checks whether the name of all the files in the folder `tests/` starts with `test_` or not.
Given this I think it may restrict the development as in signac-flow, we have python modules named `define_test_project.py`, `extract_templates.py` in the `tests/` folder.

Reply:
Yeah, on second thought this one may not be the best idea. The unit testing frameworks also enforce this by themselves to some extent by requiring the files be named a certain way to be run as tests.